src/diracx/db/jobs/status_utility.py

```diff
@@ -146,3 +150,195 @@ async def set_job_status(
         )
 
     return SetJobStatusReturn(**job_data)
+
+
+async def delete_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Delete a job by killing and setting the job status to DELETED.
+    """
+
+    await __kill_delete_job(
+        job_id,
+        config,
+        job_db,
+        task_queue_db,
+        background_task,
+    )
+
+    await set_job_status(
+        job_id,
+        {
+            datetime.now(timezone.utc): JobStatusUpdate(
+                Status=JobStatus.DELETED,
+                MinorStatus="Checking accounting",
+                StatusSource="job_manager",
+            )
+        },
+        job_db,
+        job_logging_db,
+        force=True,
+    )
+
+
+async def kill_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    await __kill_delete_job(
+        job_id,
+        config,
+        job_db,
+        task_queue_db,
+        background_task,
+    )
+
+    await set_job_status(
+        job_id,
+        {
+            datetime.now(timezone.utc): JobStatusUpdate(
+                Status=JobStatus.KILLED,
+                MinorStatus="Marked for termination",
+                StatusSource="job_manager",
+            )
+        },
+        job_db,
+        job_logging_db,
+        force=True,
+    )
+
+
+async def __kill_delete_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
+    from DIRAC.StorageManagementSystem.Client.StorageManagerClient import (
+        StorageManagerClient,
+    )
+
+    res = await job_db.search(
+        parameters=["Status", "Owner", "OwnerGroup"],
+        search=[
+            ScalarSearchSpec(
+                parameter="JobID",
+                operator=ScalarSearchOperator.EQUAL,
+                value=job_id,
+            ),
+        ],
+        sorts=[],
+    )
+    if not res:
+        raise NoResultFound(f"Job {job_id} not found")
+
+    status = res[0]["Status"]
+    owner = res[0]["Owner"]
+    owner_group = res[0]["OwnerGroup"]
```

Comment:
I find indexing into tuples like this kind of hard to read, how about this:
```suggestion
    job_info, = res

    status = job_info["Status"]
    owner = job_info["Owner"]
    owner_group = job_info["OwnerGroup"]
```

It also ensures we only have one result (which I know should be the case, but I'd rather be defensive and have situation like this crash instead of being tricky to debug).